tests/fixups/TraceFixup/main.cpp

```diff
@@ -191,80 +196,110 @@ void Log_ETW_UnRegister()
-            {
-                trace_calling_module = static_cast<bool>(callerConfig->as_boolean());
-            }
-
-            if (auto ignoreDllConfig = configObj.try_get("ignoreDllLoad"))
-            {
-                ignore_dll_load = static_cast<bool>(ignoreDllConfig->as_boolean());
-            }
-        }
+				}
+			}
 
-        if (wait_for_debugger)
-        {
-            psf::wait_for_debugger();
-        }
-    }
+			if (auto levels = configObj.try_get("traceLevels"))
+			{
+				g_traceLevels = &levels->as_object();
+				traceDataStream << " traceLevels : \n";
+
+				// Set default level immediately for fallback purposes
+				if (auto defaultLevel = g_traceLevels->try_get("default"))
+				{
+					traceDataStream << " default : \n" << defaultLevel->as_string().wide() << " ;";
+					g_defaultTraceLevel = trace_level_from_configuration(defaultLevel->as_string().string(), g_defaultTraceLevel);
+				}
+			}
+
+			if (auto levels = configObj.try_get("breakOn"))
+			{
+				g_breakLevels = &levels->as_object();
+				traceDataStream << " breakOn : \n";
+
+				// Set default level immediately for fallback purposes
+				if (auto defaultLevel = g_breakLevels->try_get("default"))
+				{
+					traceDataStream << " default : " << defaultLevel->as_string().wide() << " ;";
+					g_defaultBreakLevel = trace_level_from_configuration(defaultLevel->as_string().string(), g_defaultBreakLevel);
+				}
+			}
+
+			if (auto debuggerConfig = configObj.try_get("waitForDebugger"))
+			{
+				traceDataStream << " waitForDebugger : " << static_cast<bool>(debuggerConfig->as_boolean()) << " ;";
+				wait_for_debugger = static_cast<bool>(debuggerConfig->as_boolean());
+			}
+
+			if (auto traceEntryConfig = configObj.try_get("traceFunctionEntry"))
+			{
+				traceDataStream << " traceFunctionEntry : " << static_cast<bool>(traceEntryConfig->as_boolean()) << " ;";
+				trace_function_entry = static_cast<bool>(traceEntryConfig->as_boolean());
+			}
+
+			if (auto callerConfig = configObj.try_get("traceCallingModule"))
+			{
+				traceDataStream << " traceCallingModule : " << static_cast<bool>(callerConfig->as_boolean()) << " ;";
+				trace_calling_module = static_cast<bool>(callerConfig->as_boolean());
+			}
+
+			if (auto ignoreDllConfig = configObj.try_get("ignoreDllLoad"))
+			{
+				traceDataStream << " ignoreDllLoad : " << static_cast<bool>(ignoreDllConfig->as_boolean()) << " ;";
+				ignore_dll_load = static_cast<bool>(ignoreDllConfig->as_boolean());
+			}
+
+
+			TraceLoggingWrite(
+				g_Log_ETW_ComponentProvider,
+				"TraceFixupConfigdata",
+				TraceLoggingWideString(traceDataStream.str().c_str(), "TraceFixupConfig"),
+				TraceLoggingBoolean(TRUE, "UTCReplace_AppSessionGuid"),
+				TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+				TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA));
+
+		}
+
+		if (wait_for_debugger)
+		{
+			psf::wait_for_debugger();
+		}
+
+	}
+
+	if (reason == DLL_PROCESS_DETACH)

```

Comment:
else if

Reply:
Fixed!

Reply:
Elseif should be here (line 299) else if (reason == DLL_PROCESS_DETACH)

---
In reply to: [285207970](https://github.com/microsoft/MSIX-PackageSupportFramework/pull/41#discussion_r285207970) [](ancestors = 285207970)